platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/camera/CameraAnimatorActivity.java

```diff
@@ -116,20 +138,82 @@ public void onAnimationUpdate(ValueAnimator animation) {
     return tiltAnimator;
   }
 
-  private static class LatLngEvaluator implements TypeEvaluator<LatLng> {
+  //
+  // Interpolator examples
+  //
 
-    private final LatLng latLng = new LatLng();
+  @Override
+  public boolean onCreateOptionsMenu(Menu menu) {
+    getMenuInflater().inflate(R.menu.menu_animator, menu);
+    return true;
+  }
 
-    @Override
-    public LatLng evaluate(float fraction, LatLng startValue, LatLng endValue) {
-      latLng.setLatitude(startValue.getLatitude()
-        + ((endValue.getLatitude() - startValue.getLatitude()) * fraction));
-      latLng.setLongitude(startValue.getLongitude()
-        + ((endValue.getLongitude() - startValue.getLongitude()) * fraction));
-      return latLng;
+  @Override
+  public boolean onOptionsItemSelected(MenuItem item) {
+    if (mapboxMap == null) {
+      return false;
+    }
+    findViewById(R.id.fab).setVisibility(View.GONE);
+
+    // reset camera to init position
+    mapboxMap.moveCamera(CameraUpdateFactory.newCameraPosition(
+      new CameraPosition.Builder()
+        .target(START_LAT_LNG)
+        .zoom(11)
+        .bearing(0)
+        .tilt(0)
+        .build()
+    ));
+
+    // play animation
+    Animator animator = createExamepleInterpolator(item.getItemId());
+    if (animator != null) {
+      animator.start();
+    }
+
+    return super.onOptionsItemSelected(item);
+  }
+
+  private Animator createExamepleInterpolator(int menuItemId) {
+    switch (menuItemId) {
```

Comment:
What about removing the `switch` using a map/dictionary? `AnimatorSet`s could be created beforehand and inserted into a map using the different menu ids as values.